[13.0][ADD] hr_recruitment_notification: subscribe users to new applications automatically

joao-p-marques opened the pull request:
New module that provides a new message subtype to allow users to be automatically subscribed to new applications from a Job Position.

@Tecnativa
TT30413

ping @pedrobaeza @Yajo

yajo commented on hr_recruitment_notification/data/message_subtypes.xml:
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2021 Tecnativa - João Marques -->
+<odoo>
+    <data noupdate="1">
```
It seems to me like this shouldn't be noupdate.

joao-p-marques replied:
Well, fine by me, it just seems that that is the standard in all Odoo message subtypes. For example, in a runbot with all core modules installed:

![image](https://user-images.githubusercontent.com/38977934/131295837-ebba3d40-a9c6-4b10-a919-478e64fa51da.png)

yajo replied:
Still missing.

pedrobaeza replied:
They are noupdate in core, and they should be that way for allowing any modification (for example, `default` flag).

yajo replied:
Ah that makes sense.

yajo commented on hr_recruitment_notification/data/message_subtypes.xml:
```diff
@@ -0,0 +1,15 @@
+            <field name="description">Applicant created</field>
+            <field name="res_model">hr.job</field>
+            <field name="default" eval="False" />
+            <field name="parent_id" eval="ref('hr_recruitment.mt_applicant_new')" />
```
```suggestion
            <field name="parent_id" ref="hr_recruitment.mt_applicant_new" />
```

yajo reviewed (changes requested):
Please open PR for upstream odoo master about this.

joao-p-marques committed:
[ADD] hr_recruitment_notification: subscribe users to new applications automatically

New module that provides a new message subtype to allow users to be automatically subscribed to new applications.

TT30413

joao-p-marques commented:
Thanks @Yajo!

Changes made and PR opened to Odoo in https://github.com/odoo/odoo/pull/75736

pedrobaeza commented:
/ocabot merge nobump
